server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/ReplacePreProcessorListener.java

```diff
@@ -0,0 +1,178 @@
+import static org.eclipse.lsp.cobol.core.model.ErrorSeverity.ERROR;
+
+/**
+ * ANTLR listener, which deals only with the REPLACE compiler directives. So, that the rest of the
+ * pre-processing works on the replaced source.
+ */
+@Slf4j
+public class ReplacePreProcessorListener extends CobolPreprocessorBaseListener
+    implements GrammarPreprocessorListener {
+  @Getter private final List<SyntaxError> errors = new ArrayList<>();
+  private final ReplacingService replacingService;
+  private final List<Pair<String, String>> replacingClauses;
+  private final BufferedTokenStream tokens;
+  private final MessageService messageService;
+  private final String documentUri;
+  Deque<StringBuilder> textAccumulator = new ArrayDeque<>();
+
+  public ReplacePreProcessorListener(
+      ReplacingService replacingService,
+      MessageService messageService,
+      BufferedTokenStream tokens,
+      String documentUri,
+      List<Pair<String, String>> replacingClauses) {
+    this.replacingService = replacingService;
+    textAccumulator.push(new StringBuilder());
+    this.tokens = tokens;
+    this.messageService = messageService;
+    this.documentUri = documentUri;
+    this.replacingClauses = replacingClauses;
+  }
+
+  @Override
+  public Deque<StringBuilder> getTextAccumulator() {
+    return textAccumulator;
+  }
+
+  @Override
+  public ExtendedDocument getResult() {
+    if (!replacingClauses.isEmpty()) {
+      replace();
+    }
+    return new ExtendedDocument(accumulate(), null, null, null);
+  }
+
+  private void replace() {
+    String content = applyReplacing(pop(), replacingClauses);
+    replacingClauses.clear();
+    if (getTextAccumulator().isEmpty()) push();
+    write(content);
+  }
+
+  @Override
+  public void enterReplaceAreaStart(@NonNull CobolPreprocessor.ReplaceAreaStartContext ctx) {
+    if (!replacingClauses.isEmpty()) {
+      replace();
+    }
+    push();
+  }
+
+  @Override
+  public void enterReplacePseudoText(CobolPreprocessor.ReplacePseudoTextContext ctx) {
+    push();
+  }
+
+  @Override
+  public void exitReplacePseudoText(CobolPreprocessor.ReplacePseudoTextContext ctx) {
+    if ((ctx.getParent() instanceof CobolPreprocessor.ReplaceByStatementContext)) {
+      @NonNull
+      ResultWithErrors<Pair<String, String>> clauseResponse =
+          replacingService.retrievePseudoTextReplacingPattern(read());
+      if (clauseResponse.getErrors().isEmpty()) {
+        replacingClauses.add(clauseResponse.getResult());
+      } else {
+        clauseResponse
+            .getErrors()
+            .forEach(
+                error -> {
+                  Locality locality = retrievePosition(ctx);
+                  errors.add(
+                      SyntaxError.syntaxError()
+                          .severity(ERROR)
+                          .suggestion(
+                              messageService.getMessage(
+                                  clauseResponse.getErrors().get(0).getSuggestion()))
```

Comment:
Why do you use the first error suggestion for all the errors?

Reply:
It was a blunder. Thanks, done.